python/python/res/simulator/batch_simulator_context.py

```diff
@@ -58,25 +59,31 @@ def results(self):
         when the simulator was created the results will be returned as:
 
 
-          [  {"CMODE" : [1,2,3], "order" : [1,1,3]},
-             {"CMODE" : [1,4,1], "order" : [0,7,8]},
-             {"CMODE" : [6,1,0], "order" : [0,0,8] ]
+          { 0: {"CMODE" : [1,2,3], "order" : [1,1,3]},
```

Comment:
It is not transparent what the keys of the result correspond to. They are indeed the indices in the list of control values passed to the BatchSimulator, but this needs to be documented! In addition, I find the inconsistency in simulation ids a bit odd, could we either make them implicit or explicit in both ends; that is both when passed to start and when returning the results?

Personally I would prefer to have them explicit on both sides, where each simulation passed to start could be a dict specifying geo_realization, simulation_id and input/controls (we should perhaps get rid the control notion in libres as a BatchSimulator is definitively useful outside the scope of OR/optimization.